src/resources/uart_esp32_hal.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2023 Toitware ApS.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; version
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
+#pragma once
+#if defined(__FREERTOS__)
+
+#include <hal/uart_types.h>
+
+// This file is purely here to enable calling some lowlevel C-code defined in header files
+// in ESP-IDF. For documentation of almost all the functions, please refer to hal/uart_hal.h.
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+  
+typedef enum {
+  UART_TOIT_INTR_RXFIFO_FULL  = 0, // The RxFifo is over its threshold
+  UART_TOIT_INTR_TXFIFO_EMPTY,     // The TxFifo is under its threshold
+  UART_TOIT_INTR_PARITY_ERR,       // Parity error
+  UART_TOIT_INTR_RXFIFO_OVF,       // RxFifo overflow, not emptied fast enough
+  UART_TOIT_INTR_TX_BRK_DONE,      // Transfer break completed
+  UART_TOIT_INTR_RX_TIMEOUT,       // The RxFifo has data, not over its threshold, but some time since last byte
+  UART_TOIT_ALL_INTR_MASK,         // All interrupt bits
```

Comment:
```suggestion
  UART_TOIT_INTR_RXFIFO_FULL  = 0,  // The RxFifo is over its threshold.
  UART_TOIT_INTR_TXFIFO_EMPTY,      // The TxFifo is under its threshold.
  UART_TOIT_INTR_PARITY_ERR,        // Parity error.
  UART_TOIT_INTR_RXFIFO_OVF,        // RxFifo overflow, not emptied fast enough.
  UART_TOIT_INTR_TX_BRK_DONE,       // Transfer break completed.
  UART_TOIT_INTR_RX_TIMEOUT,        // The RxFifo has data, not over its threshold, but some time since last byte.
  UART_TOIT_ALL_INTR_MASK,          // All interrupt bits.
```